### PR TITLE
Disable fast path by default for TransformerEncoderLayer

### LINKED ISSUE #158012: TransformerEncoderLayer precision loss when fast path is enabled
### 🐛 Describe the bug

When `model.eval()` is called, usually fast path is enabled for `torch.nn.TransformerEncoderLayer` during inference. I expect the inference result is the same if `model.train()` is called, but it's not.
This indicates that the output of eval mode has severe precision loss.

This is an example:
```
import torch
from torch.nn import TransformerEncoderLayer
model = TransformerEncoderLayer(512,8)
randin = torch.randn(32,10,512)
model.eval()
oeval = model(randin)
model.train()
otrain = model(randin)

assert torch.isclose(oeval, otrain).all()
```

### Versions

torch==2.7.1+cpu
python==3.13.2
numpy==2.2.5

### PR DESCRIPTION
**Pull request by chaihahaha**
It seems the fast path implementation (flash attention) in `torch.nn.TransformerEncoderLayer` has serious precision loss.

Fixes https://github.com/pytorch/pytorch/issues/158012

### CODE REVIEW

**Comment by chaihahaha** [comment]
It seems the precision loss is because of layer norm, not fast path.
